lib/onfido/configuration.rb

```diff
@@ -30,22 +31,19 @@ def logger
       RestClient.log ||= NullLogger.new
     end
 
-    def region
-      return unless api_key
-
-      first_bit = api_key.split("_")[0]
-
-      return if %w(live test).include?(first_bit)
+    def endpoint
+      region_host = region ? region_hosts[region.downcase] : 'api.onfido.com'
+      unless region_host
+        raise "The region \"#{region.downcase}\" is not currently supported"
+      end
 
-      first_bit
+      "https://#{region_host}/#{api_version}/"
     end
 
-    def endpoint
-      if region
-        "https://api.#{region}.onfido.com/#{api_version}/"
-      else
-        "https://api.onfido.com/#{api_version}/"
-      end
+    private
+
+    def region_hosts
+      @region_hosts ||= YAML.load_file(File.join(__dir__, './config/region_hosts.yml'))
```

Comment:
ugh this repo used to be YAML free! can't we just us a hash?

Reply:
We can - I'm not really sure on the best practice of doing config in ruby projects
Where would you suggest we put the hash?

Reply:
Okay, I've moved it over to be a constant, in the module

Reply:
I think YAML probably is best practice. But it feels like overkill here.

You can either put it here or put it in a separate config file.